questions.md

```diff
@@ -4,26 +4,82 @@
    jobs you are running on the cluster?
+   
+   __You can figure out how many processors are required, RAM needed and files written by the job by prefixing the program command you are running on the cluster with ```/usr/bin/time -v```. To estimate this before running a job to get an idea of the resources you should request, you can perform this same operation on a smaller set of input data than you would run in your actual job, and then scale the resource information it provides up (i.e. use "Percent of CPU this job got", "Maximum resident set size (kbytes)", and "File system outputs" from the run on a smaller set of data to estimate processsor, RAM, and file requirements needed when the real job is run.__
+   
 3. In order to be a "good citizen", you need to have some idea of much RAM your job requires.  In particular, you need
    to know the "peak" (i.e., maximum) RAM required at any point during execution.  Show an example of the shell command
-   that you would use on a Linux machine to measure run time and peak ram usage of an arbitrary command, where the time/peak RAM values are written to a file.
+   that you would use on a Linux machine to measure run time and peak ram usage of an arbitrary command, where the time/peak RAM values    are written to a file.
+   
+   __An example of a shell command that measures and records run time and peak ram usage to file would be:__
+   
+   ```{ /usr/bin/time -v command >/dev/null;} 2>&1 | grep -E "Maximum|Elapsed" > memtime.txt``` Where "command" is the arbritrary command being run.
+   
+   __If we were to do this with a fasta file containing all _D. melanogaster_ chromosome sequences, the command and results would look like this__:
+   
+    
+    ```
+    { /usr/bin/time -v grep "A" dmel-all-chromosome-r6.24.fasta >/dev/null;} 2>&1 | grep -E "Maximum|Elapsed" > memtime.txt
+    less memtime.txt
+    Elapsed (wall clock) time (h:mm:ss or m:ss): 0:00.27
+    Maximum resident set size (kbytes): 2088
+    ```
+    
 4. What are the units of your answer for number 3?
+
+   __The units for peak RAM usage (i.e. Maximum resident set size) is kilobytes; units for run time are minutes:seconds__
+
 5. What are the bash commands for the following operations:
 
     * Checking that a file exists
+    
+    __In the directory of interest:__
+    
+    ```ls filename```
+    
     * Checking that a file exists and is not empty
+    
+    __In the directory of interest:__
+    
+    ```ls -l filename```
 
 6. How would you use the commands from your answer to 5 to write a work flow for HPC that only runs a job if the
    expected output file is **not** present.
 
+   __Using ls to generate a list of files in a directory, you could see whether a file exists and, if not, run a job with the following work flow:__
+
+   ```ls | if [ ! -e filename ]; then echo "The file is not present; the job will be run"; fi```
```

Comment:
here i would just do if [ ! -e test.txt ] instead of using ls. a one liner would look like this:
[ -e test.txt ] || CMD